libraries/AP_HAL_Linux/RCOutput_PCA9685.cpp

```diff
@@ -224,17 +217,17 @@ void RCOutput_PCA9685::push()
         *d++ = length >> 8;
     }
 
-    if (!_i2c_sem->take_nonblocking()) {
+    if (!_dev->get_semaphore()->take_nonblocking()) {
         return;
     }
 
-    hal.i2c->writeRegisters(_addr,
-                            PCA9685_RA_LED0_ON_L + 4 * (_channel_offset + min_ch),
-                            (max_ch - min_ch) * 4,
-                            &data[min_ch * 4]);
-
-    _i2c_sem->give();
+    size_t len = (max_ch - min_ch) * 4 + 1;
+    uint8_t transfer_data[len];
+    transfer_data[0] = PCA9685_RA_LED0_ON_L + 4 * (_channel_offset + min_ch);
+    memcpy(&transfer_data[1], &data[min_ch * 4], (len - 1) * sizeof(uint8_t));
```

Comment:
Nice, thanks. Only one question regarding the last line: should I change the type to uint16_t?